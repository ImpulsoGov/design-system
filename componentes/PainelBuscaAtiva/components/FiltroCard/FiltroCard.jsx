import React, { useState } from "react";
import style from "./FiltroCard.module.css";

export const FiltroCard = ({
  id,
  label = "",
  filtroID,
  handleCheckbox = () => {},
  initiallyChecked = false,
})=>{
  const [checked, toggleChecked] = useState(initiallyChecked);
  const handleChange = (event) => {
    const { id: property, checked } = event.target;
    toggleChecked(checked);
    handleCheckbox(property, checked);
  }

  return(
    <div className={style.FiltroCard} data-testid="FiltroCard">
      <input
        className={style.InputFiltroCard}
        type="checkbox"
        onChange={handleChange}
        name={filtroID}
        checked={checked}
        id={id}
      />
      <label
        className={style.LabelFiltroCard}
        data-testid="FiltroCardLabel"
        htmlFor={id}
      >
        {label}
      </label>
    </div>
  )
}
