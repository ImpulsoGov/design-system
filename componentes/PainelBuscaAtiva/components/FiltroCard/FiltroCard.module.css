.FiltroCard{
  display: flex;
  flex-direction: row;
  font-family: 'Inter';
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  gap: 10px;
  background: #EFF5F9;
  border-radius: 10px;
  width: fit-content;
  padding: 0 6px;
  margin-left: 15px;
}

.InputFiltroCard{
  accent-color: #2EB280;
  color: white;
}

.InputFiltroCard:checked::before{
  color: white !important;
}

.LabelFiltroCard {
  padding: 12px 0;
}