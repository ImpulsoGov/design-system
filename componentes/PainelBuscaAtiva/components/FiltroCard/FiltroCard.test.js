import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { FiltroCard } from './FiltroCard';

const COMPONENT_NAME = 'FiltroCard';
const scenarios = [
  { label: 'Tainá', filtroID: 'acs_nome' },
  { label: 'Alessandra Santos', filtroID: 'acs_nome' },
  { label: '3', filtroID: 'acs_nome' },
];

describe(`Componete: ${COMPONENT_NAME}`, () => {
  it('deve renderizar corretamente', async () => {
    const component = render(<FiltroCard { ...scenarios[0] } />);
    const filtroCard = component.getByTestId(COMPONENT_NAME);

    expect(filtroCard).toMatchSnapshot();
  });

  it('deve exibir um checkbox marcado após o click', async () => {
    const user = userEvent.setup();
    render(<FiltroCard { ...scenarios[1] } />);
    const checkbox = await screen.findByRole('checkbox');

    await user.click(checkbox);

    expect(checkbox).toBeChecked();
  });

  it('deve exibir um checkbox desmarcado após dois clicks', async () => {
    const user = userEvent.setup();
    render(<FiltroCard { ...scenarios[2] } />);
    const checkbox = await screen.findByRole('checkbox');

    await user.click(checkbox);
    await user.click(checkbox);

    expect(checkbox).not.toBeChecked();
  });
});
