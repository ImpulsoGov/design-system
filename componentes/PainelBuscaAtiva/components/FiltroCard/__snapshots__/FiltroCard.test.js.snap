// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componete: FiltroCard deve renderizar corretamente 1`] = `
<div
  class="FiltroCard"
  data-testid="FiltroCard"
>
  <input
    class="InputFiltroCard"
    id="Tainá"
    name="acs_nome"
    type="checkbox"
  />
  <p
    data-testid="FiltroCardLabel"
  >
    Tainá
  </p>
</div>
`;
